astropy/table/__init__.py

```diff
@@ -68,5 +68,10 @@ class Conf(_config.ConfigNamespace):  # noqa
     import astropy.io.fits.connect
     import astropy.io.misc.connect
     import astropy.io.votable.connect
-    import astropy.io.misc.asdf.connect
     import astropy.io.misc.pandas.connect  # noqa: F401
+
+    # asdf_astropy is an optional dependency and may not be installed
+    try:
+        import asdf_astropy.io.connect
```

Comment:
If possible, please use the machinery at https://github.com/astropy/astropy/blob/main/astropy/utils/compat/optional_deps.py for optional dependency checks.